### ISSUE TITLE
Big red warning message into PrestaShop when not using the right PHP version

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#20862]
#### Is your feature request related to a problem?

I just saw 2 bug reports about very, very weird behaviors ... the people reporting them said "I upgraded to 1.7.6.7 and now this page does [weird behavior]". Obviously I could not reproduce the bugs.

Finally the main reason behind this very weird bug was they were not using the right PHP version.

#### Describe the solution you'd like

In order not to waste users time and PrestaShop team time, I suggest that, on Back Office pages, when we detect that user PHP version is not supported, we display a BIG RED MESSAGE telling this to user.

This will help users troubleshoot issues faster 😄 

The message would say _You are using a PHP version that is not supported and this can affect your store. Please check PHP requirements and change for the suitable PHP version._ in _Admin.Notifications.Warning_. - edited by @LouiseBonnard on September 11th, following @Julievrz's [wording suggestion](https://github.com/PrestaShop/PrestaShop/issues/20862#issuecomment-688342471).

It should be displayed only on the dashboard. Here is the design mockup:

![image](https://user-images.githubusercontent.com/32565890/99088822-d6121f00-25cc-11eb-9d8f-851745963b5a.png)

![supportedphp-alert (1)](https://user-images.githubusercontent.com/32565890/99820532-323aed00-2b51-11eb-8f42-bd865e087c4f.png)

Here are the specifications: https://github.com/PrestaShop/prestashop-specs/pull/173

**Comment by florine2623** [comment]
Ping @PrestaShop/prestashop-product-team 
Here's a great improvement suggested by @matks.
Do you think it could be implemented ?

Thanks!

**Comment by LouiseBonnard** [comment]
Nice idea, thanks @matks! I suggest displaying it every time a user goes to his store, so at each connection, on the first page displayed. A popup like a modal might be more effective since it requires an action from the user to either _Upgrade_ (redirecting to https://www.php.net/downloads?) or _Close_ the window. @TristanLDD, @prestascott, what do you recommend?

For the warning message, I propose _You are using a PHP version that is no longer supported and this can affect your store. Please upgrade to PHP %s version._ in _Admin.Notifications.Warning_. And what about What do you think, @matks and @Julievrz?

**Comment by PierreRambaud** [comment]
> Nice idea, thanks @matks! I suggest displaying it every time a user goes to his store, so at each connection, on the first page displayed. A popup like a modal might be more effective since it requires an action from the user to either _Upgrade_ (redirecting to https://www.php.net/downloads?) or _Close_ the window. @TristanLDD, @prestascott, what do you recommend?
> 
> For the warning message, I propose _You are using a PHP version that is no longer supported and this can affect your store. Please upgrade to PHP %s version._ in _Admin.Notifications.Warning_. And what about What do you think, @matks and @Julievrz?

You're considering they are using only and outdated version. What about recent versions of PHP that are not already supported?

**Comment by LouiseBonnard** [comment]
We could think about a wording that would apply to both cases then.

**Comment by Julievrz** [comment]
> We could think about a wording that would apply to both cases then.

What do you think of this sentence: " You are using a PHP version that is not supported and this can affect your store. Please check [PHP requirements](https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/#php-requirements) and change for the suitable PHP version."

**Comment by PululuK** [comment]
Hi @matks 
We also need to add an alert on system information, something like this :
![image](https://user-images.githubusercontent.com/16455155/92499630-6498a780-f1fc-11ea-875c-20fd8f0c8b98.png)

**Comment by prestascott** [comment]
Hi everyone!

Thank you for your report @matks!

Adding a modal warning message can be useful but can also be aggressive for our users. 
Like @LouiseBonnard said, I also suggest to display the modal only at the connection. And then, once he closes it, it's still displayed on the page but smaller as an alert so this information isn't lost. 

Usually, a modal contains a title, a description and buttons. I suggest to add a title like _"PHP update required"_ and a button _"Check PHP requirements"_ for example in order to help users solving their PHP version issue. 

Wdyt?

**Comment by matks** [comment]
> Hi everyone!
> 
> Thank you for your report @matks!
> 
> Adding a modal warning message can be useful but can also be aggressive for our users.
> Like @LouiseBonnard said, I also suggest to display the modal only at the connection. And then, once he closes it, it's still displayed on the page but smaller as an alert so this information isn't lost.
> 
> Usually, a modal contains a title, a description and buttons. I suggest to add a title like _"PHP update required"_ and a button _"Check PHP requirements"_ for example in order to help users solving their PHP version issue.
> 
> Wdyt?

I'm good with this 👍 

Can I use the "standard modal display" we use, for example, for "bulk delete" actions ?

So if I try to make a summary;

1. When user logins into BO and the shop is running a PHP version not compatible, and he reaches the dashboard, we display the modal. User can either click on the "Check PHP requirements" CTA or close it.

2. Whether he closes the modal or clicks the CTA, we close the modal. We dont reopen the modal as long as user stays connected. However we have a warning message inside Dashboard.

3. If user logs out then logs-in, we re-trigger item 1: we show the modal.

4. Following @PululuK suggestion, the warning displayed inside Dashboard can be displayed into "System informations" BO page too, with php version being displayed in red to attract user attention.

**Comment by prestascott** [comment]
I'm good with your summary, thanks! :)

**Comment by LouiseBonnard** [comment]
It's all okay for me except the first point:

> When user logins into BO and the shop is running a PHP version not compatible, and he reaches the dashboard, we display the modal. User can either click on the "Check PHP requirements" CTA or close it.

Note that the user does not necessarily reach the dashboard when he/she logs in so I assume the modal must be displayed at each connection, whatever the page reached. 

Also, in some cases, the Welcome module could be enabled, which displays popping up onboarding journey. It might be difficult to display the two so if we need to set up an order of preference, I'm in favor of displaying the PHP modal before the onboarding journey.

**Comment by PululuK** [comment]
@matks Instead of modal, I think we need to display a message above all the admin pages. Something like :
![image](https://user-images.githubusercontent.com/16455155/93074587-e2d5cd80-f684-11ea-87c4-97b532dd2145.png)

**Comment by matks** [comment]
> It's all okay for me except the first point:
> 
> > When user logins into BO and the shop is running a PHP version not compatible, and he reaches the dashboard, we display the modal. User can either click on the "Check PHP requirements" CTA or close it.
> 
> Note that the user does not necessarily reach the dashboard when he/she logs in so I assume the modal must be displayed at each connection, whatever the page reached.

I know but this is a lot harder to do 😞 .
"When user logs in on BO for the 1st time and reaches Dashboard, display the modal" => it's alright, scope is quite clear,, easy to implement and to test (for QA team)

"When user logs in on BO for the 1st time and reaches a page, whatever BO page, display the modal" => there's legacy pages, there's migrated pages, there's BO pages added by modules. I expect the 3 of them to require a different implementation. Moreover I expect some specific pages (Product Page ? Carrier Page ?) to be quite unique and require even more extra work. On some pages also the rendering of the modals might integrate badly into the pageAnd this means the "How to test" for QA will be "well, it must on ... all the BO pages". Quite a big QA scope to say the least 😅 .

This is why I suggest focusing on Dashboard only: it will be so much easier to implement and test. And I expect the user, at some point, to reach the Dashboard 😄 so even if he 1) logs-in 2) goes to Product Page 3) goes to Dashboard I should be able to "detect" he reaches Dashboard and _there_ display the modal.

> Also, in some cases, the Welcome module could be enabled, which displays popping up onboarding journey. It might be difficult to display the two so if we need to set up an order of preference, I'm in favor of displaying the PHP modal before the onboarding journey.

Indeed one more obstacle on the path 😅 I'll see what I can do, but since this is a module it'll be hard to control how it pops up from the Core.

**Comment by matks** [comment]
> @matks Instead of modal, I think we need to display a message above all the admin pages. Something like :

That is not mine to decide as I am no Product Manager, I'll let @LouiseBonnard follow up then 😉

**Comment by PierreRambaud** [comment]
> @matks Instead of modal, I think we need to display a message above all the admin pages. Something like :
> ![image](https://user-images.githubusercontent.com/16455155/93074587-e2d5cd80-f684-11ea-87c4-97b532dd2145.png)

I agree, it will be easier to integrate and also better, don't want to receive a modal alert, each time I open a new page :sweat_smile:

**Comment by prestascott** [comment]
@PululuK It's a nice idea but this component doesn't exist. I guess we can create it but it won't be red and will be yellow since it's a warning. 
I still suggest to add a modal instead that only appears when users log in. 

Here is what it would look like as an alert: 
![stickyalert](https://user-images.githubusercontent.com/54888056/93106398-825c8580-f6b0-11ea-8157-e442c96b627d.png)

**Comment by PululuK** [comment]
> @PululuK It's a nice idea but this component doesn't exist. I guess we can create it but it won't be red and will be yellow since it's a warning.
> I still suggest to add a modal instead that only appears when users log in.
> 
> Here is what it would look like as an alert :

@prestascott An alert already exists on this page :
![Capture du 2020-09-16 15-34-27](https://user-images.githubusercontent.com/16455155/93345105-bd86c200-f832-11ea-908b-20aa614fcf76.png)

But the warning idea looks good to me !

**Comment by prestascott** [comment]
@PululuK I'm glad that the warning idea looks good to you! 👍 

And yes, in fact there's already an alert on this page but this is an old design that we're trying to move on. 
This Dashboard page hasn't been migrated yet that's why we prefer using our new components such as the alert that I added.

**Comment by LouiseBonnard** [comment]
Right! I discussed with @prestascott and, indeed, the warning solution is less aggressive to users than the modal window so we've decided to drop the first idea and focus on the warning notification. Our job is to warn the users that they will have difficulties if they do not use the right PHP version. Once this message is sent, it is up to the users to get a suitable PHP version.

Here are the specifications, it's ready for review: https://github.com/PrestaShop/prestashop-specs/pull/173
